### ISSUE TITLE
transformation docs.

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#1064]
The docs for the transformation module aren't very good. The python module has some good descriptions but the c-module has only some short sentences. We should synchronize the two docs and maybe extend them to give some context to allow a better usage for a novice user (or experienced since I spend some time today reading to docs and code to figure out what I need to do analyze rotations).

I found this [transformation3d](https://github.com/matthew-brett/transforms3d) package. They are based on the same source we use. Their docs are very extensive and I think we can copy some of them. This is useful for the euler angle functions. Because the default "euler angle" notation we use is not the standard euler angle notation.

**Comment by mnmelo** [comment]
I agree with using/adapting a more up-to-date transformation library, comment and code-wise.
Perhaps transformation3d also addresses the scaling inconsistencies in #962?

**Comment by richardjgowers** [comment]
@mnmelo I'd love to move transformations outside of mda into its own package, the problem with that package is it only uses the python version of the source we use (not cython).

The C/Python docs shouldn't be two sets of docs anyway, iirc the C functions just shadow the Python ones, so there needn't be different docs.

**Comment by mnmelo** [comment]
Hadn't seen that transformation3d was python-only. Then disregard the consistency remark I made earlier.

**Comment by kain88-de** [comment]
Why do we need a c implementation of that library anyway? We can load a lot of work on them. Meaning when I use this I still have to call functions a lot which is slow. My gut feeling is that the c version isn't that much different in speed for practical use.

**Comment by kain88-de** [comment]
> My gut feeling is that the c version isn't that much different in speed for practical use.

I just did some quick comparison. The C-version is a lot faster. It will matter in loops but for one time applications the python version is still fast enough with micro second execution time.
